modules/kargoAnalyticsAdapter.js

```diff
@@ -1,14 +1,71 @@
+import { _each, logError } from '../src/utils.js';
+import { ajax } from '../src/ajax.js';
 import adapter from '../src/AnalyticsAdapter.js';
 import adapterManager from '../src/adapterManager.js';
 
-var kargoAdapter = adapter({
-  analyticsType: 'endpoint',
-  url: 'https://krk.kargo.com/api/v1/event/prebid'
-});
+const EVENT_URL = 'https://krk.kargo.com/api/v1/event';
+const KARGO_BIDDER_CODE = 'kargo';
+const CONSTANTS = require('../src/constants.json');
+
+const analyticsType = 'endpoint';
+
+let _logData = {
+  auctionTimeout: 0,
+  bidId: '',
+};
+
+var kargoAnalyticsAdapter = Object.assign(
+  adapter({ analyticsType }), {
+    track({ eventType, args }) {
+      if (typeof args !== 'undefined') {
+        switch (eventType) {
+          case CONSTANTS.EVENTS.AUCTION_INIT: {
+            _logData.auctionTimeout = args.timeout;
+            break;
+          }
+          case CONSTANTS.EVENTS.BID_TIMEOUT: {
+            handleTimeout(args);
+            break;
+          }
+        }
+      }
+    }
+  }
+);
+
+function handleTimeout (timeouts) {
+  let sent = false;
+  _each(timeouts, timeout => {
+    if (timeout.bidder === KARGO_BIDDER_CODE && !sent) {
+      _logData.auctionId = timeout.auctionId;
+
+      sendTimeoutData(_logData);
+      sent = true;
+    }
+  });
+}
+
+function sendTimeoutData (data) {
+  try {
+    ajax(
+      `${EVENT_URL}/timeout`,
+      null,
+      {
```

Comment:
The Kraken endpoint by @er-san also expects an `rt` field for the Prebid Response Time (not 100% sure on what this means). Will that be added in the future?

Reply:
@wwongkargo unfortunately, it doesn't look like we're going to be able to calculate response time from the client-side. After we hit the timeout, the request gets cut off completely, so we don't know how long it would have taken us to respond. We can leave the parameter for future timeout endpoints with other adapters.